bevy_lint/src/lints/pedantic/borrowed_reborrowable.rs

```diff
@@ -167,14 +161,13 @@ impl<'tcx> LateLintPass<'tcx> for BorrowedReborrowable {
                 continue;
             };
 
-            // This lint would emit a warning on `&mut self` if `self` was reborrowable. This isn't
-            // useful, though, because it would hurt the ergonomics of using methods of
-            // reborrowable types.
-            //
-            // To avoid this, we skip any parameter named `self`. This won't false-positive on
-            // other function arguments named `self`, since it is a special keyword that is
-            // disallowed in other positions.
-            if arg_names[arg_index].is_some_and(|ident| ident.name == kw::SelfLower) {
+            // If the argument is named `self`, skip it. Without this check the lint would be
+            // emitted for `&mut self` if `self` was reborrowable, which isn't wanted! That would
+            // just be annoying for engine developers trying to add useful methods to reborrowable
+            // types.
+            if let PatKind::Binding(_, _, ident, _) = body.params[arg_index].pat.kind
+                && ident.name == kw::SelfLower
```

Comment:
This does the same thing as the old code, it just retrieves the argument names a different way.